src/components/screens/HistoryScreen/HistoryItem.tsx

```diff
@@ -28,21 +29,28 @@ const HistoryItem: React.FC<HistoryItemProps> = ({
   const { themeColors } = useColors();
   const [isLoading, setIsLoading] = useState(true);
   const [historyItem, setHistoryItem] = useState<any>(null);
+  const [historyItemUi, setHistoryItemUi] = useState<any>(null);
 
   // Load history item data on component mount or when dependencies change
   useEffect(() => {
     const buildHistoryItem = async () => {
       try {
         const historyItemData = await mapHistoryItemData({
           operation,
-          accountBalances,
           publicKey,
           networkDetails,
-          network,
-          themeColors,
         });
 
+        const historyItemUiData = mapHistoryItemUi(
```

Comment:
And we can pass the data to a helper function that we maintain in this codebase which will create JSX markup around our data